test/framework/resource/virtualservice/manager.go

```diff
@@ -40,8 +42,18 @@ type defaultManager struct {
 func (m *defaultManager) WaitUntilVirtualServiceActive(ctx context.Context, vs *appmesh.VirtualService) (*appmesh.VirtualService, error) {
 	observedVS := &appmesh.VirtualService{}
 	return observedVS, wait.PollImmediateUntil(utils.PollIntervalShort, func() (bool, error) {
-		if err := m.k8sClient.Get(ctx, k8s.NamespacedName(vs), observedVS); err != nil {
-			return false, err
+
+		// sometimes there's a delay in the resource showing up
```

Comment:
I remember we did something similar for VN and VR but we shouldn't need this change isn't it? "wait.PollImmediateUntil" should keep on trying the func until it returns true.I don't see a reason for this change, what am I missing?

Reply:
`PollImmediateUntil` tries the func until it returns true or error so in this case `Get` returns an err that virtualservice is not found. So we need an additional check to re-try on error because of eventual consistency. I'll adjust this to align better so `PollImmediateUntil` can retry instead of doing it within the func